apps/communications/contacts/views/details/js/details_ui.js

```diff
@@ -0,0 +1,463 @@
+          // Sanity check
+          if (utils.mozContact.haveEmptyFields(currentAddress,
+              ['streetAddress', 'postalCode', 'locality', 'countryName'])) {
+            continue;
+          }
+          var address = currentAddress.streetAddress || '';
+          var escapedStreet = Normalizer.escapeHTML(address, true);
+          var locality = currentAddress.locality;
+          var escapedLocality = Normalizer.escapeHTML(locality, true);
+          var escapedType = Normalizer.escapeHTML(currentAddress.type, true);
+          var country = currentAddress.countryName || '';
+          var escapedCountry = Normalizer.escapeHTML(country, true);
+          var postalCode = currentAddress.postalCode || '';
+          var escapedPostalCode = Normalizer.escapeHTML(postalCode, true);
+
+          var addressField = {
+            streetAddress: escapedStreet,
+            postalCode: escapedPostalCode,
+            locality: escapedLocality || '',
+            countryName: escapedCountry,
+            type: this._(escapedType) || escapedType ||
+              TAG_OPTIONS['address-type'][0].value,
+            'type_l10n_id': currentAddress.type,
+            i: i
+          };
+          var template = utils.templates.render(this.addressesTemplate,
+            addressField);
+          this.listContainer.appendChild(template);
+        }
+      });
+    },
+
+    renderDates: function(contact) {
+      if (!contact.bday && !contact.anniversary) {
+        return;
+      }
+
+      var dateString = '';
+
+      var fields = ['bday', 'anniversary'];
+      var l10nIds = ['birthday', 'anniversary'];
+
+      var rendered = 0;
+      for (var j = 0; j < fields.length; j++) {
+        var value = contact[fields[j]];
+        if (!value) {
+          continue;
+        }
+        try {
+          dateString = utils.misc.formatDate(value);
+        } catch (err) {
+          console.error('Error parsing date');
+          continue;
+        }
+        var element = utils.templates.render(this.datesTemplate, {
+          i: ++rendered,
+          date: dateString,
+          type: this._(l10nIds[j])
+        });
+
+        this.listContainer.appendChild(element);
+      }
+    },
+
+    renderNotes: function(contact) {
+      if (!contact.note || contact.note.length === 0) {
+        return;
+      }
+      var container = document.createElement('li');
+      var title = document.createElement('h2');
+      title.setAttribute('data-l10n-id', 'comments');
+      container.appendChild(title);
+      for (var i = 0; i < contact.note.length; i++) {
+        var currentNote = contact.note[i];
+        var noteField = {
+          note: Normalizer.escapeHTML(currentNote, true) || '',
+          i: i
+        };
+        var template = utils.templates.render(this.notesTemplate, noteField);
+        container.appendChild(template);
+        this.listContainer.appendChild(container);
+      }
+    },
+
+    renderSocial: function(contact) {
```

Comment:
Should we remove this, as we are not going to support FB, or will be that work be done in the specific bug?